sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -249,4 +251,28 @@ public static String bomAwareToString(byte[] bytes, String contentType) {
             }
         }
     }
+
+    /**
+     * Retrieves the application ID from either a {@link ClientOptions} or {@link HttpLogOptions} for use in a
+     * {@link UserAgentPolicy}.
```

Comment:
nit: don't need to specify that it's used in `UserAgentPolicy`. Application id can be used for any purpose and this method shouldn't be tied to it's usage.

Reply:
Removed